core/src/test/java/de/jplag/BasicFunctionalityTest.java

```diff
@@ -87,66 +87,28 @@ void testPartialPlagiarism() throws ExitException {
         assertEquals(12, biggestMatch.get().matches().size());
     }
 
-    // TODO SH: Methods like this should be moved to the API and also should accept wildcards
-    private double getSelectedPercent(JPlagResult result, String nameA, String nameB) {
-        return getSelectedComparison(result, nameA, nameB).map(JPlagComparison::similarity).orElse(-1.0);
-    }
-
-    private Optional<JPlagComparison> getSelectedComparison(JPlagResult result, String nameA, String nameB) {
-        return result.getAllComparisons().stream()
-                .filter(comparison -> comparison.firstSubmission().getName().equals(nameA) && comparison.secondSubmission().getName().equals(nameB)
-                        || comparison.firstSubmission().getName().equals(nameB) && comparison.secondSubmission().getName().equals(nameA))
-                .findFirst();
-    }
-
+    /**
+     * This test case uses single files as submissions, which is a valid scenario.
+     */
     @Test
-    void testMultiRootDirNoBasecode() throws ExitException {
-        List<String> paths = List.of(getBasePath("basecode"), getBasePath("SimpleDuplicate")); // 3 + 2 submissions.
-        JPlagResult result = runJPlag(paths, it -> it);
-        assertEquals(5, result.getNumberOfSubmissions());
-    }
+    void testSingleFileSubmisssions() throws ExitException {
+        JPlagResult result = runJPlagWithDefaultOptions("SimpleSingleFile");
 
-    @Test
-    void testMultiRootDirSeparateBasecode() throws ExitException {
-        String basecodePath = getBasePath("basecode-base");
-        List<String> paths = List.of(getBasePath("basecode"), getBasePath("SimpleDuplicate")); // 3 + 2 submissions.
-        JPlagResult result = runJPlag(paths, it -> it.withBaseCodeSubmissionDirectory(new File(basecodePath)));
-        assertEquals(5, result.getNumberOfSubmissions());
-    }
+        assertEquals(2, result.getNumberOfSubmissions());
+        assertEquals(1, result.getAllComparisons().size());
+        assertEquals(1, result.getSimilarityDistribution()[6]);
+        assertEquals(0.6207, result.getAllComparisons().get(0).similarity(), DELTA);
 
-    @Test
-    void testMultiRootDirBasecodeInSubmissionDir() throws ExitException {
-        String basecodePath = getBasePath("basecode", "base");
-        List<String> paths = List.of(getBasePath("basecode"), getBasePath("SimpleDuplicate")); // 2 + 2 submissions.
-        JPlagResult result = runJPlag(paths, it -> it.withBaseCodeSubmissionDirectory(new File(basecodePath)));
-        assertEquals(4, result.getNumberOfSubmissions());
-    }
+        var matches = result.getAllComparisons().get(0).matches();
+        var expectedMatches = runJPlagWithDefaultOptions("SimpleDuplicate").getAllComparisons().get(0).matches();
```

Comment:
Hard to see that "SimpleDuplicate" contains the same files as "SimpleSingleFile". Maybe add some explaining comment either to the files themselves or here in code?
Also the method assumes that matches are ordered which is not guaranteed, right? So maybe do some sorting or replace the pairwise iteration by an equalsSize and forEachExists logic.